webknossos/webknossos/dataset/dataset.py

```diff
@@ -1339,6 +1352,62 @@ def add_layer_from_images(
                     "[WARNING] Some images are larger than expected, smaller slices are padded with zeros now. "
                     + f"New bbox is {layer.bounding_box}, expected {expected_bbox}."
                 )
+
+            # Check if category of layer is set correctly
```

Comment:
please add a comment that the first heuristic is very simple and that the second one more thorough. however, it can only happen after the actual conversion which is why these are two passes. something along those lines..